test/verify/machineslib.py

```diff
@@ -178,13 +178,41 @@ def prepareStorageDeviceOnISCSI(m, target_iqn, orig_iqn=None):
 class TestMachines(NetworkCase):
     created_pool = False
     provider = None
+    tmp_storage = "/mnt/libvirt"
 
     def setUp(self):
         # HACK: fix this to get along with NetworkCase.setUp(), and use super()
         MachineCase.setUp(self)
+        m = self.machine
+
+        # Prepare tmp directory
+        m.execute("mkdir {0}".format(self.tmp_storage))
+        self.addCleanup(m.execute, "rm -rf {0}".format(self.tmp_storage))
+
+        # Keep pristine state of libvirt
+        orig = "/var/lib/libvirt"
+        backup = os.path.join(self.tmp_storage, "libvirt.backup")
+        m.execute("cp -a {0} {1}".format(orig, backup))
+        self.addCleanup(m.execute, "rm -rf {0} && cp -a {1} {0}".format(orig, backup))
+
+        # Cleanup domains definitions
+        etc_orig = "/etc/libvirt"
+        etc_backup = os.path.join(self.tmp_storage, "etc_libvirt.backup")
+        m.execute("cp -a {0} {1}".format(etc_orig, etc_backup))
+        self.addCleanup(m.execute, "rm -rf {0} && cp -a {1} {0}".format(etc_orig, etc_backup))
+
+        # Cleanup pools
+        self.addCleanup(m.execute, "rm -rf /var/run/libvirt/storage/*")
```

Comment:
Just /run, please. /var/run hasn't been a real thing for like a decade now. Also, is there anything in /run/libvirt/ which needs to be kept? If libvirt.service is stopped, I can't think of anything. /run is a tmpfs, so after boot it's guaranteed to be empty.

Reply:
I think so, it was giving me problem when I tried to destroy(archive) the whole `/run/libvirt` so I just picked parts that were important for us.